### PR TITLE
depend on gitMinimal instead of git

### PR DESCRIPTION
**Pull request by Mic92**
git requires requires in graphics libraries dependencies etc.

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by zimbatm** [comment]
We used to have something like that, but it led to having two copies of git since other programs include git in their runtime closure.

File: nixos/server/default.nix [hunk 1]
```diff
@@ -14,6 +14,9 @@
     ../../shared/server.nix
   ];
 
+  # used in shared/server.nix
+  config.programs.git.package = lib.mkDefault pkgs.gitMinimal;
```
Comment (zimbatm) [inline nixos/server/default.nix:18]:
```suggestion
  programs.git.package = lib.mkDefault pkgs.gitMinimal;
```

**Review by zimbatm — APPROVED** [review]
let's find out

**Comment by Mic92** [comment]
> let's find out

For met it was the other way around. I had gitMinimal first in my closure. But in any case, users now can configure which git version they want.

**Pull request merged**